library/packages/src/lib/packages/package_downloader.rb

```diff
@@ -0,0 +1,81 @@
+# encoding: utf-8
+# ------------------------------------------------------------------------------
+# Copyright (c) 2017 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+
+require "yast"
+
+require "tempfile"
+require "shellwords"
+
+require "yast2/execute"
+
+Yast.import "Pkg"
+
+module Yast2
+  # Downloads a package from a known package manager (libzypp) repository.
+  #
+  # @note For downloading files outside of a libzypp repository use the
+  # FileFromUrl::get_file_from_url method:
+  # https://github.com/yast/yast-installation/blob/fba82c3c9abfc44e3d31c8658bf96079d74e0298/src/lib/transfer/file_from_url.rb#L89
+  #
+  # @example Downloading a package
+  #   begin
+  #     downloader = PackageDownloader.new(3, "yast2")
+  #     tmp = Tempfile.new("downloaded-package-")
+  #     downloader.download(tmp.path)
+  #     # do something with the package...
+  #   ensure
+  #     tmp.close
+  #     tmp.unlink
+  #   end
+  #
+  class PackageDownloader
```

Comment:
Why not use the `Packages` namespace? Classes defined in that directory are under the `Packages` too. Searching for another example, `library/general/src/lib/installation/` uses the `Installation` module. I would prefer being consistent.